### ISSUE TITLE
 Multiple thumbnails generated when saving custom template

### ISSUE THREAD

**Issue by LuckynaSan** [swissspidy/web-stories-wp#12805]
## Bug Description
When saving a custom page template, the plugin takes a screenshot of it as a preview thumbnail. 

A user has reported that multiple images have been created. 

Support topic: https://wordpress.org/support/topic/saved-template-pagination/

## Additional Context

- Plugin Version: 1.28.0
- WordPress Version:
- Operating System:
- Browser:

**Comment by swissspidy** [comment]
Previously: #12513

**Comment by miina** [comment]
I've been trying to repro it locally but without luck so far...
It does happen in the QA environment though.

What I've found out so far:
- It seems to generate a new image every time the saved template is rendered
- Locally, when debugging `SavedPageTemplate`, the image generation seems to happen just once. Since the image already exists then the following prevents creating a second image with `shouldPostBlob` being `false`: https://github.com/GoogleForCreators/web-stories-wp/blob/40780ca7c61909a70c73cbd4e1952ec4f6c18572/packages/story-editor/src/components/library/panes/pageTemplates/savedPageTemplate.js#L139-L140
- In the QA environment, the `page.image.url` is falsy, meaning that the image will be re-generated. `page.image` value for each template in the QA environment in the `SavedPageTemplate` component:
<img width="445" alt="Screenshot 2022-12-22 at 20 08 19" src="https://user-images.githubusercontent.com/3294597/209240347-59fde5d2-f2dc-44d9-ae08-308dfd0bc91b.png">

@swissspidy Any thoughts on why this might be happening in QA but not locally? (Not expecting an answer before the new year due to holidays but leaving it here anyway, will continue debugging tomorrow regardless)

**Comment by swissspidy** [comment]
For context, the falsey value is coming from here:

https://github.com/GoogleForCreators/web-stories-wp/blob/40780ca7c61909a70c73cbd4e1952ec4f6c18572/packages/wp-story-editor/src/api/pageTemplate.js#L43-L50

(Aside: we should stop using these empty objects for non-existent stuff, it's confusing)

This actually led me down a path of figuring this out.

Here's what I've found:

1. `getCustomPageTemplates()` loads all saved templates from the server. Since the poster images are stored as featured media in WP, we "embed" the featured media object so that it's included in the response object under `_embedded.wp:featuredmedia`.
2. The above code looks up the image info in `_embedded.wp:featuredmedia` and sets that for use in the editor. If it's missing, we get that empty object.
3. The editor checks whether an image is set or not, and creates a new one if needed, calling `updatePageTemplate` to save the featured image ID.
4. `updatePageTemplate()` worked as expected. The poster was uploaded and its ID associated with the page template.

So since the upload actually worked as expected, the culprit was in `getCustomPageTemplates()`.

In essence, that function makes a call like `await wp.apiFetch({ path: 'web-stories/v1/web-story-page?_web_stories_envelope=true&_fields=id%2Cstory_data%2Cbody%2Cheaders&_embed=wp:featuredmedia' })` to get only the needed fields for all the templates (ID, story data, body, headers).

-> **Note:** the "special" `_embedded` field is not listed in `_fields`. That's because it should be automatically included, according to this code in WP core:

https://github.com/WordPress/wordpress-develop/blob/0ae94f9a665fa61ef500e3adeb080748c86564fc/src/wp-includes/rest-api/endpoints/class-wp-rest-controller.php#L599-L606

Turns out, however, that this is not working for this request. No idea why 🤷  The field is definitely automatically included for other requests.

So when loading all the page templates, `_embedded.wp:featuredmedia` was never actually included, leading the editor to think that there were simply no featured images.

This definitely used to work, so I assume it's one of the unexpected new behaviors in WP 6.1... Looking at the code above in WP I'd even call it a bug.

Maybe @spacedmonkey can shed some light on that in the new year.

Anyway, #12872 should now fix this.
